ice/recipes/finalN.py

```diff
@@ -0,0 +1,395 @@
+
+List briefly the primary population(s) that were included in this study. ALL population(s) should be mutually exclusive and be the primary population(s) that were included in this study.
+
+-Healthy individuals
+
+Below is an abstract from a randomized control trial(RCT)
+
+Abstract: {abstract}
+
+List briefly the primary population(s) that were included in this study. ALL population(s) should be mutually exclusive and be the primary population(s) that were included in this study.
+
+-"""
+
+
+def extract_numbers(text: str) -> list[str]:
+    # Only extracts numbers written in digits so that the int function can convert them to ints.
+    words = text.split()
+
+    number_strings = []
+    numbers_set = set("0123456789")
+    number_strings += list(
+        filter(lambda x: set(x).intersection(numbers_set) != set(), words)
+    )
+
+    # Remove parentheses
+    remove_parentheses = (
+        lambda s: s.replace("(", "")
+        .replace(")", "")
+        .replace("...", "")
+        .replace("..", "")
+    )
+    number_strings = list(map(remove_parentheses, number_strings))
+
+    # Remove "," or "." from the end of the number string
+    for i, number in enumerate(number_strings):
+        if number[-1] == "," or number[-1] == ".":
+            number_strings[i] = number[:-1]
+
+    return number_strings
```

Comment:
```suggestion
    numbers_set = set("0123456789")
    return [
        word.replace("(", "")
            .replace(")", "")
            .replace("...", "")
            .replace("..", "")
            .rstrip(",.")
        for word in words
        if numbers_set.intersection(word)
    ]
```

1. This assumes that removing more than one trailing `,` or `.` is OK, which your original code does not.
2. Regexes or an actual library might be better suited here.
3. Is it fine to return strings containing numbers that clearly aren't numbers, like `"agley-2021.pdf"`?